File: languages/go/entity/identity/credential_status.go
package identity

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// CredentialStatus represents the status information for a credential, including its ID, revocation nonce, and type.
type CredentialStatus struct {
	Id              string
	RevocationNonce int64
	Type            string
}

func NewCredentialStatusFromProto(s *proto.CredentialStatus) CredentialStatus {
	if s == nil {
		return CredentialStatus{}
	}
	return CredentialStatus{
		Id:              s.Id,
		RevocationNonce: s.RevocationNonce,
		Type:            s.Type,
	}
}

func (c CredentialStatus) ToProto() *proto.CredentialStatus {
	return &proto.CredentialStatus{
		Id:              c.Id,
		RevocationNonce: c.RevocationNonce,
		Type:            c.Type,
	}
}

File: languages/js/src/entity/identity/credential_status.ts
import * as identityEntitiesProto from "../../bridge/proto/identity_entities";

/**
 * Represents the status information for a credential, including its ID, revocation nonce, and type.
 */
export class CredentialStatus {
  id: string;
  revocationNonce: number;
  type: string;

  /**
   * Constructs an CredentialStatus object with the specified parameters.
   * @param id 
   * @param revocationNonce 
   * @param type 
   */
  constructor(id: string, revocationNonce: number, type: string) {
    this.id = id;
    this.revocationNonce = revocationNonce;
    this.type = type;
  }

  public toProto(): identityEntitiesProto.CredentialStatus {
    return identityEntitiesProto.CredentialStatus.fromPartial({
      id: this.id,
      revocationNonce: this.revocationNonce,
      type: this.type
    });
  }

  static fromProto(
    r: identityEntitiesProto.CredentialStatus
  ): CredentialStatus {
    return new CredentialStatus(r.id, r.revocationNonce, r.type);
  }
}
